feat: Strength Ratings Changes

Blogaugis opened the pull request:
Changes the strength rating system from 1-6 to 1-10. Still Work-In-Progress.
Current notable tweaks:

- Reordered the Sisters roster (faction 5 should be between faction 2 (imperials) and 6 (the elves)).
- Changed the "Negligible"-"Overwhelming" naming to "Squad"-"Army Corps", as per contemporary OoB formation naming.
- Added commented out code for future Imperium, Admech and Inquisition roster additions.
- Added comments.
- Tweaked the Sisters Strength in ratings 1-3.

Blogaugis committed:
Strength Ratings Changes

Changes the strength rating system from 1-6 to 1-10.
Still Work-In-Progress.

EttyKitty commented:
> Changed the "Negligible"-"Overwhelming" naming to "Squad"-"Army Corps", as per contemporary OoB formation naming.

An army corp of tyranids sounds a bit funny. Maybe consider more ambiguous names?

Also, perhaps consider adding a number at the end, for ease of access. So that the end result is `Tyranids (Overwhelming 9)`.

*Update - tbh, I'm not sure if removing the word completely and leaving only the number is a bad call. I.e. `Tyranids (9)`.
*Update2 - probably is a bad call, as new players may not know that 10 is max and so have no point of reference.

OH296 commented:
Not sure i'm so keen on the idea of moving to such specific terms for the description of enemy forces. It's somewhat constraining and less universal. 
Looking past matters of taste this work will need to be ported to work with #179 as the position of this logic has been moved to the create sequence

OH296 commented on objects/obj_drop_select/Draw_0.gml:
```diff
@@ -47,12 +50,16 @@ if (attacking=11){fork=csm;tt1="CSMs (";}
+if (fork=7) then tt2="Regiment";
+if (fork=8) then tt2="Brigade";
+if (fork=9) then tt2="Division";
+if (fork>=10) then tt2="Army Corps";
```
```suggestion
var _size_terms = ["", "Squad","Demi-Platoon","Platoon","Demi-Company","Company","Battalion","Regiment","Brigade","Division","Army Corps"];
if (fork<array_length(_size_terms)){
     tt2 = _size_terms[fork];
}
```

This is a far more concise and preferable way of writing this

OH296 replied:
as another point if this area is being reworked can the variable name tt2 be changed to something that means something

OH296 replied:
also as a way of doing this number can then be added dynamically e.g ```var _final_term = $"{_size_terms} ({fork})" ```

Blogaugis replied:
Ok, I tried to implement the suggested change, github didn't allow to implement this with outdated labels.

OH296 replied:
noted, it's the general conscpet that's important potentially accessing these terms is going to happen a lot in game code so they should saved in an array as displayed, this will save much time for you in the conversion o moving to 10 or more stages of strength. It also means that the code is becoming more concise and better than the code which duke originally made rather than being more bloated

OH296 commented on objects/obj_drop_select/Draw_0.gml:
```diff
@@ -315,12 +325,16 @@ repeat(20){q+=1;
-        if (fork=4) then tt2="High";
-        if (fork=5) then tt2="Very High";
-        if (fork>=6) then tt2="Overwhelming";
+        if (fork=1) then tt2="Squad";
```
same as above in general no reason for this logic to be repeated twice as another point

OH296 commented on objects/obj_ncombat/Alarm_0.gml:
```diff
@@ -280,11 +280,11 @@ if (battle_special = "space_hulk") {
 	if (enemy = 7) {
 		modi = random_range(0.80, 1.20) + 1;
-		make = round(max(3, player_starting_dudes * modi));
+		make = round(max(3, player_starting_dudes * modi)); // Should we really leave this functionality, forcing the player to pick as few marines as possible? I don't think this makes sense
```
```suggestion
		make = round(max(3, player_starting_dudes * modi)); //makes enemy forces in space hulks scale with big player forces
``` 
comments such as these can be brought up here rather than in the code, code comments are generally for points of clarification on functions , otherwise the code ends up cluttered. as is this line is the only thing that makes space hulks unique to normal encounters

Blogaugis replied:
Ok.
Personally, I am against the mechanics that artificially inflate the number of enemy forces.
If the strategic map says it has strength 4 forces, then it should have strength 4 forces in battle.
The only changes to that should be the equipment they have (whether it is randomization or something more complex, of the % of heavy weapons-carrying troops and other specialists).

I can see the deception and similar tactics, if we do get more nuanced intelligence gathering, but for now, the only faction that should be able to engage in deception should be chaos, daemons, perhaps inquisition, and the 3 hi-tech ones of Necrons, Eldar and, arguably, T'au.

OH296 replied:
I'm not sure i entirely see how this mechanic "Artificially inflates the number of enemy forces" by having an element of dynamism while i admit thats what the mechanic does from a purely mathmatical perspective that's not an entirely constructive way of looking at a problem in some senses imo. 
The mechanics purpose is to potentially humble a player by sending in large forces (reason for this below).
point taken in terms of the feeling of being deceived but as the intention of the code is not meant to represent being deceived but rather to simulate the idea that a large contingent may alert the large forces of enemy normally unaware of your small exploration it seems your main gripe would be solved via a one line wording of "small groups may easily explore the <term> without causing alarm however it is harder for larger groups to move without detection".  Or some such terms.

Blogaugis replied:
So, it's an infiltration/object of interest extraction mission.
This should mean that the enemy strength rating should not change upon completing mission with infiltration approach. Or, if it does change, then change very little.

I'd prefer to give the player an option to clear out the space hulk, though such an affair would be costly, but potentially very rewarding - such as getting new ship(s).

Still, the player should be informed that the rules of engagement in Space hulks (and perhaps dungeons and ruins) are different from the typical battle on the ground or space.

OH296 commented:
As a small point i would like to point out tha for this work to be handled properly there are in excess of probably 50 but possibly over 100 files that need to be augmented for this rework to function correctly. Be aware of this as simply changing the numbers in the two currently augmented scripts will not be a sufficient threshold of conversion for merge

Blogaugis committed:
Tweaks 12-10

- Added numbers to the strength rating names;
- Changed position of descriptive comments;
- Reworked SoB strengths 2-4; SOBs begin to receive support (Chaff) units at 4 strength.

To be decided if the names need (re)tweaks.

Blogaugis commented:
> [...]consider adding a number at the end[...]

Done.

> It's somewhat constraining and less universal.

Fair. I do think we may need to use custom naming for different factions, but this may be a bit out of scope for this.

Blogaugis commented on objects/obj_ncombat/Alarm_0.gml:
```diff
@@ -576,6 +576,258 @@ if (enemy = 2) {
+		u.dudes[3] = "Retributor"; // Should be made random whether it is 1 retributor or 1 dominion, maybe
+		u.dudes_num[3] = 1;
+		u.dudes[4] = "Dominion";
+		u.dudes_num[4] = 1;
```
As a side note: this kind of asks for randomization stuff or more complex conditions:
It can instead be the surviving Retributor or Dominion sisters, Repentias as well.
Also perhaps it could have a Rhino or 2, indicating a small mechanized detachment, instead or regular.
Though, the complexity is probably a bit out of scope for this, so I'll probably experiment with randomization instead.

Blogaugis replied:
On the other hand, I think it is a bit out of scope for more advanced functions at this time. Maybe in a future PR.

OH296 commented:
> Not sure i'm so keen on the idea of moving to such specific terms for the description of enemy forces. It's somewhat constraining and less universal. Looking past matters of taste this work will need to be ported to work with #179 as the position of this logic has been moved to the create sequence

also take note of this your work needs to be rebased onto the current state of 0.10 otherwise the conflicts will be hell later

Blogaugis committed:
Changes 12-11 1

- Tweaked the strength ratings 4-6 of SoBs.

Blogaugis committed:
Update objects/obj_ncombat/Alarm_0.gml

Comment change.

Co-authored-by: Nelsonh <81228864+OH296@users.noreply.github.com>

Blogaugis committed:
Update Draw_0.gml

Attempting to implement a suggestion.

Blogaugis commented:
> also take note of this your work needs to be rebased onto the current state of 0.10 otherwise the conflicts will be hell later

I don't see the option to do so, at the moment.
I'll try to, when I see it.

EttyKitty committed:
Merge branch 'release/0.10.0.0' into 0.10.0-StrengthRatings

EttyKitty committed:
Post-merge migration

Blogaugis committed:
Changes 12-12

- Adjusted ratings 4-6 again, to ensure that the troop numbers match, in case leader spawns;
- SoB OoB is considered complete.
May still need some tweaks, but testing is needed for this.

Blogaugis committed:
Changes 12-13

- Added a comment for Gretchin stats;
- Added the new Ork faction strength ratings 1-10.
TODO: Test if game allows these numbers to exist in a single lane (4000+ troops).

Addendum: I skipped the Elves for now. With lack of land combats with them, I am not sure of their capabilities.

Blogaugis committed:
Changes 12-14

Reworked the 1-3 T'au strength ratings.

Blogaugis committed:
Changes 12-16

- Tau faction implemented.

Note: It currently is the smallest faction in terms of numbers of ratings, but that should change once Skeletons and Elves are added.

Blogaugis committed:
Changes 12-19

- Reworked 1-3 strength ratings for Tyranids.
TODO - decide how to balance event battles, discuss Nid progression in discord.

Blogaugis committed:
Changes 12-20

- Reworked 1-4 strength ratings of Tyranids, there are now 2 types:
- The Genestealer Cultist formations (currently commented out);
- And Tyranid formations, which are the bioforms for battle. Until the rest of logic for picking the right formation type is implemented, this is going to be the default formation types used.

Blogaugis committed:
Changes 12-21

- Reworked strength rating 5 for Tyranids, both variants.

Blogaugis committed:
Changes 12-22

- Reworked/added the 6-10 strength ratings of Space Bugs (both variants).

Note: I am not confident of Bug capabilities, as they are less numerous than greenskins. Balance adjustments may be required.

EttyKitty committed:
docs: Add discord badge to readme (#233)

- Add a Discord badge to the README.
- Fix a typo.

Co-authored-by: sourcery-ai[bot] <58596630+sourcery-ai[bot]@users.noreply.github.com>

EttyKitty committed:
docs: Add discord badge to readme (#233)

- Add a Discord badge to the README.
- Fix a typo.

Co-authored-by: sourcery-ai[bot] <58596630+sourcery-ai[bot]@users.noreply.github.com>

OH296 committed:
improve error handling, log repairs

OH296 committed:
forge queue visual error with table

OH296 committed:
comparing arrays fix

Blogaugis committed:
Work on Space Bugs 12-26

- Tweaked the Nid strength 3, and buffed ratings 4-10, both variants.

EttyKitty committed:
ci: Test clone releases action

EttyKitty committed:
ci: Test clone releases action

OH296 committed:
fix : Solve visual errors in forge (#322)

- solves visual error with forge queue with manufactoring research
- add more error handling

## Summary by Sourcery

Bug Fixes:
- Fix visual overlap with research projects in the forge queue display.

OH296 committed:
fix : draw unit image comparing arrays not possible  (#323)

## Summary by Sourcery

Bug Fixes:
- Fix the condition to check for non-empty bionic options array using
array_length instead of direct comparison.

OH296 committed:
trying to round a string

OH296 committed:
PR: Fix trying to round a string in ui_manage (#324)

## Summary by Sourcery

Bug Fixes:
- Prevent error when displaying experience value if it is not a number.

OH296 committed:
move specialist point handler intialisation point

EttyKitty committed:
ci: Update access key name

EttyKitty committed:
ci: Update access key name

OH296 committed:
partial solve to icon crash issue

OH296 committed:
PR: Fix Initialisation sequences in load script causing crashes  (#326)

## Summary by Sourcery

Bug Fixes:
- Prevent crashes caused by missing chapter icon sprites during loading.

OH296 committed:
role not initialised correctly in add man

OH296 committed:
PR: Fix role not being initialised correctly in scr_add_man (#327)

## Summary by Sourcery

Bug Fixes:
- Fixed a bug where the unit's role was not initialised correctly when
adding a new man.

OH296 committed:
fix: Use of safer string interpolation (#328)

- Prevent potential bugs when deleting save files by using string interpolation instead of string concatenation.

EttyKitty committed:
fix: A couple of scr_image_cache errors (#330)

- Fix loading custom chapter images by handling cases where the image path is not set.
- Fix the wrong array push target.
- Format the function.

EttyKitty committed:
ci: Try write-all premission to fix tag move

EttyKitty committed:
ci: Replace write-all premission with a WORKFLOW_TOKEN

At this point, I'm out of ideas on how to fix this.

OH296 committed:
custom work arounds

OH296 committed:
PR: Fix Custom Livery for heads to match departments (#334)

as reported here
https://discord.com/channels/714022226810372107/1317251464242135132/1321969505575895040
custom livery for department heads is wrong
## Summary by Sourcery

Bug Fixes:
- Fixed an issue where heads did not have the correct livery.

OH296 committed:
PR: Fix Fleet retreating crash (#335)

- fixes crash with fleet retreating
- solves a crash with reallocation of marines on ships
- solves minor error in en_fleet alarm 1 from wrong scoping

## Summary by Sourcery

Implement fleet retreating logic and fix a crash caused by reallocating
marines on ships.

Bug Fixes:
- Fix a crash that occurred when a fleet retreated.
- Fix an issue where marines were not correctly reallocated when ships
were destroyed during a retreat.

Enhancements:
- Implement a new `player_retreat_from_fleet_combat` function to handle
fleet retreat logic.

OH296 committed:
PR: Fix minor scoping error in one of the new ship return events (#336)

## Summary by Sourcery

Bug Fixes:
- Fix an issue where piety was not correctly used when calculating
corruption for returned ships.

OH296 committed:
PR: Fix bug from later Initialisation of local complex_set variable moved in draw_unit_image (#338)

## Summary by Sourcery

Bug Fixes:
- Move the initialization of the local variable 'complex_set' to the
beginning of the 'scr_draw_unit_image' function to ensure it is properly
initialized before use.

EttyKitty committed:
docs: Test addition of a PR template (#329)

- Add a pull request template with sections for description, reasons, and related links.

EttyKitty committed:
docs: Test addition of a PR template (#329)

- Add a pull request template with sections for description, reasons, and related links.

OH296 committed:
PR : Fix possible over running of array length with ship weapons (#337)

## Summary by Sourcery

Bug Fixes:
- Prevent crashes caused by mismatched array lengths when accessing ship
weapons and their facing directions.

---------

Co-authored-by: sourcery-ai[bot] <58596630+sourcery-ai[bot]@users.noreply.github.com>

EttyKitty committed:
ci: Add PR title validation workflow (#341)

- Validate pull request titles using the `thehanimo/pr-title-checker` GitHub Action.

EttyKitty committed:
ci: Add PR title validation workflow (#341)

- Validate pull request titles using the `thehanimo/pr-title-checker` GitHub Action.

OH296 committed:
feat(balance): Tweaks to grav weapon as per play testing (#339)

Increase the damage of the Grav-Pistol and add maintenance values for grav weapons.

OH296 committed:
PR: Stop bug where Squad View no longer appears (#340)

- attempts to fix the common bug of the squad view becoming inaccessible
- it appears it was due to a clause making it non viewable if instances
of obj_temp3 existed
- attempt to remove as many uneeded obj_temp3 references as possible
that can be done safely and easily

## Summary by Sourcery

Bug Fixes:
- Fix a bug that prevented the Squad View from appearing when instances
of obj_temp3 existed.

VanWeapon committed:
fix: Custom role saving (#345)

- Fixes custom roles not saving during custom chapter creation.

VanWeapon committed:
fix: Homeworld save/load for custom chapters (#347)

- Fixes an issue where all loaded custom chapters had a hive world due to a typo in scr_save_chapter.

OH296 committed:
Sprite: New Complex Components for Tartaros Armour (#342)

- Tartaros right and left leg components
- Tartaros chest components
- new complex area key "gorget"
- Tartaros gorget components

I'm no artist this is as good as i can do i'm afraid

KRdaMystic committed:
PR: Create Speacial Traits for first founding Chapters (#344)

<!--- You can add `@sourcery-ai` into the title, so that the bot
auto-generates a title -->
## Description of changes
-Adds starting traits for marines in founding chapters. 
## Reasons for changes
-Add more variance between the 9 founding chapters.
## Related links
-The
[project](https://discord.com/channels/714022226810372107/1320186099641090128)

## Summary by Sourcery

New Features:
- Introduced unique starting traits for each of the nine founding
chapters of Space Marines to increase variance and diversity between
them.

OH296 committed:
fix: Artifact continuity issue (#348)

- solves various issues with artifact continuity.
- artifacts were not being destroyed when not being recovered off of
killed marines leaving them in limbo
- occasionally the artifact in question being de-equipped due to being
gifted or some other actor was not equipped to the array loc of the
stated unit
- to solve this the de-equip function now searches for amarine with the
artifact equipped and manually de-equips it from them

## Summary by Sourcery

Fix artifact continuity issues by ensuring artifacts are correctly
de-equipped from units and destroyed when no longer recoverable.

Bug Fixes:
- Resolved issues with artifacts not being destroyed when not recovered
from killed marines.
- Fixed a bug where de-equipping an artifact due to gifting or other
actions did not correctly remove it from the unit\'s inventory.

OH296 committed:
PR : Fix drop_select not displaying in screen correctly (#349)

fixes -
https://discord.com/channels/714022226810372107/1322615907259908158

## Summary by Sourcery

Bug Fixes:
- Fix incorrect centering of the drop-down menu.

OH296 committed:
fix: Uncommon variable type error in ui_manage (#350)

https://discord.com/channels/714022226810372107/1322620181586907160
fixes this
basically just a minor type error

## Summary by Sourcery

Bug Fixes:
- Fix a type error that occurred when `selection_data` did not have a
`purpose` field.

EttyKitty committed:
fix: Captains and CM not healing (loop from 1) (#351)

Changed the loop start index from 1 to 0 in the apothecary_simple function to include all units in the healing check.

KRdaMystic committed:
PR : Adjust founding chapter trait values (#353)

## Description of changes
- Adjusts founding chapter trait values
## Reasons for changes
- Old values resulted in negative effects

## Summary by Sourcery

Enhancements:
- Updated trait values for several founding chapters.

OH296 committed:
PR: New Sprites -  Tartaros face plate, helm and trim variants (#354)

## Summary by Sourcery

New Features:
- Added Tartaros faceplate, helm, and trim variants.

---------

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

OH296 committed:
PR : Improve ordering and reduce duplicate line scr_add_man (#355)

fixes-
https://discord.com/channels/714022226810372107/1120687959365128243/threads/1322910920816721980

## Summary by Sourcery

Bug Fixes:
- Fix the order of operations when adding experience and updating the
role of a new marine.

EttyKitty committed:
PR: Fix And improve the advisor text on the recruitment screen (#343)

## Summary by Sourcery

Improve the advisor text regarding recruitment and gene-seed status.

Bug Fixes:
- Fix multiple grammar and style issues in the advisor text.
- Correct the logic for displaying the recruitment status based on the
number of marines and the recruiting status.
- Fix the logic for displaying the gene-seed status and the number of
recruiting worlds.

VanWeapon committed:
fix: Boarder sprites and fleet sprites being wrong (#358)

- Correct the display of boarder and fleet sprites by adjusting the image indexing in `scr_image` calls.

OH296 committed:
PR: Fix spawning bugs with outdated function name (#359)

- bug from outdated function name in scr_add_man
- occasionally stored unit.ship_location would go stale 
- will need to find better solution in future in theory scr_ship_clean
should prevent this

## Summary by Sourcery

Bug Fixes:
- Fixed a bug that caused marine location data to become stale.

OH296 committed:
feat(sprite): Mk8 components (#356)

- add gorget component to mk8 and make some variants
- remove gorget from base mk8 so that more mk7 helms are compatible to
be drawn with mk8
- add mk8 chest components that can extend variation with mk7 chest
components

## Summary by Sourcery

New Features:
- Added new gorget and chest pieces for the Mk 8 armour set.

---------

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

OH296 committed:
fix: Vehicle purchase (#360)

- brings the scr_add_vehicle script in line with the changes to variable
array lengths with ships

## Summary by Sourcery

Bug Fixes:
- Fix an issue with adding vehicles to companies with variable-length
ship arrays.

OH296 committed:
PR: Fix forge assignment bugs (#361)

- general overhaul of FeatureSelected struct
- correct defualt options for <unit>.is_at_location
- correct tracking of techmarine forge assignments fixed

## Summary by Sourcery

Fix forge assignment bugs and improve the FeatureSelected struct.

Bug Fixes:
- Correct default options for unit location checks.
- Correct tracking of Techmarine forge assignments.

Enhancements:
- Overhaul of the FeatureSelected struct to improve data handling.

KRdaMystic committed:
PR: Improve Balance Obliterated trait (more specialists)(balance) (#364)

gives obliterated a fighting chance.

<!--- PR title format should be `commit type: Title` -->
<!--- Commit types can be found at
https://github.com/pvdlg/conventional-commit-types?tab=readme-ov-file#commit-types
-->
<!--- You can add `@sourcery-ai` into the title, so that the bot
auto-generates a title -->
## Description of changes
<!--- Leverage the list functionality, if you have many changes -->
-Adds one specialist to each of the offices to avoid inquisition
penalties
-Strength points over 5 now count for 10 scouts in place of 5
-removed the `siege = 0` from obliterated's block of code
## Reasons for changes
<!--- Leverage the list functionality, here as well -->
- Making obliterated more playable instead of the current guaranteed
traitor declaration by the inquisitor.

## Summary by Sourcery

Rebalance the "Obliterated" trait by reducing penalties to specialist
counts and increasing the scout bonus for strength points above 5.

Bug Fixes:
- Reduce the penalties applied to specialist counts for the
"Obliterated" trait.

Enhancements:
- Increase the scout bonus from strength points over 5 for "Obliterated"
trait.

KRdaMystic committed:
PR:Allow Tactical Bikers and Marines to sabotage  when chapter are ambushers(#365)

## Description of changes
<!--- Leverage the list functionality, if you have many changes -->
-Allows Tactical Bikers to do sabotage missions
-Allows Tactical marines to do sabotage missions for chapters with
Ambushers
## Reasons for changes
<!--- Leverage the list functionality, here as well -->
-Add more to Lightning warriors and Ambushers to make them more
competetive as advantages.

## Summary by Sourcery

New Features:
- Tactical Bikers and Tactical Marines with the "Ambushers" advantage
can now perform sabotage missions.

VanWeapon committed:
PR:  Player can define extra vehicles in custom chapter json (#362)

<!--- PR title format should be `commit type: Title` -->
<!--- Commit types can be found at
https://github.com/pvdlg/conventional-commit-types?tab=readme-ov-file#commit-types
-->
<!--- You can add `@sourcery-ai` into the title, so that the bot
auto-generates a title -->
## Description of changes
<!--- Leverage the list functionality, if you have many changes -->
- implements the `extra_vehicles` attribute from chapter jsons which
works on init now. All extra vehicles are added to the 10th company so
they show up on the homeworld at the beginning of the game
## Reasons for changes
<!--- Leverage the list functionality, here as well -->
- implement unfinished feature
## Related links
<!--- Other PRs; Discord bug reports, messages, threads, etc.; outside
docs, etc.; -->
-
https://discord.com/channels/714022226810372107/714027491412410369/1323805664723734549

## Summary by Sourcery

New Features:
- Added support for the "extra_vehicles" attribute in chapter JSON files
to add vehicles to the 10th company during game initialization.

OH296 committed:
fix: Repair slaughtersong event (#370)

EttyKitty committed:
ci: Update pull_request_template (#369)

EttyKitty committed:
ci: Update pull_request_template (#369)

EttyKitty committed:
fix(Marine Spawning): Roll armor works on termies again and is improved (#368)

OH296 committed:
fix: ai_a script increase efficiency and error handling to find unresolved crash (#371)

OH296 committed:
fix: Planet and faction sprites displaying wrong sprites (#372)

OH296 committed:
fix: Crash from unset sector name (#373)

OH296 committed:
fix: Incorrect method name causing crash during Spyer fight (#374)

OH296 committed:
fix: improve ship combat checks (#375)

Blogaugis committed:
25-01-04 Chaos SR1-5

- Reworked Strength Ratings 1-5 for chaos, also rebranded them into "Chaos Undivided Cultists".

Added some TODO's for future faction flavor and variants.

VanWeapon committed:
fix: Tidy up the aspirant trial display on chapter creation slides (#377)

VanWeapon committed:
refactor: Chapter trait checking standardisation  (#376)

EttyKitty committed:
ci: A bunch of minor CI changes (#378)

EttyKitty committed:
ci: A bunch of minor CI changes (#378)

KRdaMystic committed:
fix: Weapon and formation options for Bikers and Assaults (#357)

- Fixes optional weapons for bikers and assaults to utilize weapon lists.
- Fixes formation location selection options for aforementioned squad types

EttyKitty committed:
fix: A couple of missed old trait checks (#379)

EttyKitty committed:
refactor: Wrap a ton of stuff into try/catch and format (#381)

EttyKitty committed:
fix: Variable scope in scr_add_vehicle (#385)

Co-authored-by: Nelsonh <81228864+OH296@users.noreply.github.com>

Blogaugis committed:
25-01-06 CSR5-10

- Chaos Undivided Cultist rework complete.

TODO's from earlier remain a consideration for future.

EttyKitty committed:
fix: Misplaced broken trait check (#382)

Fixed a crash that occurred on game load.

EttyKitty committed:
fix(sprite): Iron Halo display position on Artificer Armor (#383)

EttyKitty committed:
fix: Term Sergeant and Veteran broken loadouts (#388)

EttyKitty committed:
revert: Armour randomization by default (#389)

OH296 committed:
fix: Add safeguards and streamline adding ship to combat (#390)

OH296 committed:
fix: controller alarm_7 use modern ship methods (#391)

OH296 committed:
fix: move to modern method loading unloading troops controller step (#394)

OH296 committed:
fix: error handling to solve issue with sector name (#393)

OH296 committed:
fix: Typo making lava planet sprite not display (#395)

OH296 committed:
fix: Controller alarm 3 (#396)

OH296 committed:
fix: Manage hot keys interrupting company naming  (#397)

OH296 committed:
fix: PlanetData error handling to planet_forces variable (#392)

OH296 committed:
fix: Vehicle spawning (#398)

OH296 committed:
fix: Trying to call alarm on non existing object (#399)

OH296 committed:
fix: default options not being set breaking up squads loadouts (#400)

OH296 committed:
fix: add default argument for adding items (#401)

OH296 committed:
fix: typo in alarm7 obj_ncombat from earlier fix (#405)

OH296 committed:
fix: Player ship class and arrays (#406)

OH296 committed:
fix: random caching error with UnitButtonObject (#407)

OH296 committed:
fix: whelp better give it a go ruins shit (just go with it) (#408)

EttyKitty committed:
Merge branch 'stable-candidate'

VanWeapon committed:
chore: json file error (#366)

EttyKitty committed:
feat: Separate armor heads (#367)

EttyKitty committed:
refactor: First company command gear/size assign (#402)

EttyKitty committed:
feat: Edits to robes, psy hood and daemon binders (#403)

EttyKitty committed:
feat: Remove "Power Weapon" from the player side (#411)

OH296 committed:
fix: Repair old guard armour distribution system (#413)

OH296 committed:
feat: new skitarii base sprite (#409)

OH296 committed:
feat: Allow land speeders on raids (#386)

OH296 committed:
fix: get rid of restart variables to stop all related crashes (#414)

OH296 committed:
fix: wrong race variable on new recruits (#416)

EttyKitty committed:
fix(visual): Force Sword/Axe side flip (#415)

OH296 committed:
fix: Potential array issue with self-healing ships (#417)

OH296 committed:
fix: Redundant check causing an early game over (#418)

OH296 committed:
fix: odd bug from player being wiped out in battle (#419)

OH296 committed:
fix: check inquisitor in array range (#428)

OH296 committed:
fix: Use correct index for missions (#427)

OH296 committed:
fix: case typo on shrine for recruit data (#429)

OH296 committed:
fix: Wrong argument passed into system_feature_bool (#430)

OH296 committed:
fix: ai_a error improve exception message (#432)

OH296 committed:
feat: Refactor bionic legs drawing and add variant (#404)

EttyKitty committed:
feat: Wolf-Priest helmet; Chaplain helm drawing refactor (#433)

EttyKitty committed:
feat: Helmets adjustments (#434)

VanWeapon committed:
refactor: Diplomacy and advisor icon refactor to new image system (#363)

OH296 committed:
fix: Error handling to unit_finder_panel (#436)

OH296 committed:
refactor: Memory safe drawing of components (#435)

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

EttyKitty committed:
ci: More title checker triggers (#437)

OH296 committed:
fix: star_select huge rework with bug fix focus (#438)

Blogaugis committed:
CSMs 1-4

- Reworked 1-4 Strength Ratings of spikey boys.

OH296 committed:
fix: Typo causes crash when problem on planet (#442)

OH296 committed:
fix: custom_advisor splash images breaking game after save (#443)

EttyKitty committed:
fix: Change error handling of NameGenerator (#439)

OH296 committed:
fix: Move variable declaration to solve crash on drop_Select (#444)

OH296 committed:
fix: add init of custom_advisors in obj_ini creation (#445)

OH296 committed:
fix: typo causing crash when destroying gene pods (#446)

OH296 committed:
feat: Concept shaders complex livery and cloaks  (#440)

Co-authored-by: EttyKitty <20323032+EttyKitty@users.noreply.github.com>

OH296 committed:
feat: New Player system spawn options (#384)

OH296 committed:
fix: clean up a few bugs with new planet options (#447)

OH296 committed:
fix: selection of research method changed to allow ships to be selected (#448)

OH296 committed:
fix: many misc sprite bugs (#449)

VanWeapon committed:
fix: Attempted fix for crash on image id 0 (#450)

VanWeapon committed:
fix: Minor alignment issues with save chapter button (#451)

OH296 committed:
Fix: gloriana vessels work again (#453)

VanWeapon committed:
fix: Diplo splash images for trade dialogue (#454)

OH296 committed:
fix: Colonists and introduce PurchaseButton constructor (#456)

OH296 committed:
fix: Make dungeon battles and local battles run on modern roster system (#459)

OH296 committed:
fix: Issue wrong  spawn name would spawn for home planet (#460)

OH296 committed:
feat: Blood Angels Complex livery (#461)

VanWeapon committed:
fix: Random exit from creation screen on click (#462)

Blogaugis committed:
25-02-03

- 5-10 ratings reworked.
Special encounter battles and fights with World Eaters still need ajustments.

Blogaugis committed:
2025-02-04

- Reworked the 1-4 World Eaters CSM variant strength ratings.

Blogaugis committed:
2025-02-05

- Reworked the 5-10 WE variant strength rating roster.

Blogaugis committed:
2025-02-06

- Daemon strength ratings 1-10 reworked.

MCPO-Spartan-117 committed:
fix: `Ambushers`+`Lightning Warriors` crash (#464)

MCPO-Spartan-117 committed:
fix: Homeworld and recruiting world names mixing up (#465)

MCPO-Spartan-117 committed:
fix: Tech Heresy detection var in wrong scope (#466)

Blogaugis committed:
2025-02-07

- 1-10 Strength Rating roster for space skeletons reworked.

MCPO-Spartan-117 committed:
feat: If population equals zero then reset all influence (#468)

MCPO-Spartan-117 committed:
feat: Make Purge interaction with Cults more obvious (#467)

Blogaugis committed:
2025-02-08 Elves

- Space Elves fully reworked;
- Strength ratings 7-10 are semi-placeholders, as I just moved the original 3-6 ratings to their place. Should be good enough for now;
- Also fixed a few errors, with names having wrong letters.

EttyKitty committed:
feat: New sprites for Adeptas (#469)

frost923 committed:
fix: Rhino missing HK missile armament (#472)

EttyKitty committed:
feat: Add pasting support for cheats (#470)

MCPO-Spartan-117 committed:
feat: Make Bombard purging reduce corruption (#473)

EttyKitty committed:
feat: Rework how debugl is used and works (#476)

MCPO-Spartan-117 committed:
fix: Garrison fixes and minor changes (#475)

Blogaugis committed:
Compile fix 1

Fixes 1 of compile errors.

Blogaugis committed:
Compile fix 2

Fixes remaining compile errors.

Blogaugis committed:
Pre-Final fixes

- It compiles and can be played.
- Needs fixing rewards for SR7-10.

MCPO-Spartan-117 committed:
fix: Remove tau influence from corruption cals (#479)

MCPO-Spartan-117 committed:
feat: Custom and more event colors (#478)

Co-authored-by: Nelsonh <Humes101@gmail.com>

Blogaugis committed:
Final Commit, pre-rebase

- Fixes the crash on 10SR battle end.
Issues detected:
- SR rating normally does not increase beyond 6 (Company) level.

Blogaugis committed:
Merge remote-tracking branch 'upstream/main' into 0.10.0-StrengthRatings

Blogaugis committed:
Final touches

- Adding the `option 2` of the threat level (Strength Rating) names, as, in my opinion, the best option (option 2 in community vote).

Blogaugis commented:
I'm going to close this one here, and open it on the `main` of chapter master repository.
